CONTRIBUTING.md

```diff
@@ -1,5 +1,74 @@
-# CONTRIBUTING
+# Contributing to Mistral AI Client for Dart
 
-## Development flow
+Thank you for considering contributing to [Your Project Name]! We appreciate your time and effort.
 
-## How to build
+## Table of Contents
+
+- [Contributing to Mistral AI Client for Dart](#contributing-to-mistral-ai-client-for-dart)
+  - [Table of Contents](#table-of-contents)
+  - [How Can I Contribute?](#how-can-i-contribute)
+    - [Reporting Bugs](#reporting-bugs)
+    - [Suggesting Enhancements](#suggesting-enhancements)
+    - [Code Contributions](#code-contributions)
+    - [Documentation](#documentation)
+  - [Getting Started](#getting-started)
+    - [Setting Up the Development Environment](#setting-up-the-development-environment)
+  - [Style Guide](#style-guide)
+  - [Review Process](#review-process)
+
+## How Can I Contribute?
+
+### Reporting Bugs
+
+Before submitting a bug report, please check if the issue has already been reported in the [issue tracker](https://github.com/nomtek/mistralai_client_dart/issues). If it hasn't, create a new issue with a clear title and description, including steps to reproduce the bug.
+
+### Suggesting Enhancements
+
+If you have an idea for an improvement or a new feature, we'd love to hear about it! Create a new issue in the [issue tracker](https://github.com/nomtek/mistralai_client_dart/issues) with your suggestion, and we can discuss it further.
+
+### Code Contributions
+
+We welcome contributions to the codebase. To contribute:
+
+1. [Fork the repository](https://help.github.com/en/github/getting-started-with-github/fork-a-repo).
+2. Create a new branch for your feature or bug fix.
+3. Make your changes and commit them with clear and concise messages.
+4. Push your changes to your fork.
+5. [Submit a pull request](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/proposing-changes-to-your-work-with-pull-requests), explaining the changes you made.
+
+### Documentation
+
+Improving documentation is a valuable contribution. If you find areas that need clarification or have ideas for better explanations, please make the necessary updates.
+
+## Getting Started
+
+### Setting Up the Development Environment
+
+- Check [pubspec.yaml](https://github.com/nomtek/mistralai_client_dart/blob/main/pubspec.yaml) for any dependencies and versions required to build project.
+
+## Style Guide
+
+No particular style guide. We are using [very_good_analysis](https://pub.dev/packages/very_good_analysis) package to keep code consistent.
```

Comment:
It's probably worth adding a style guide in the end 😉 Could you create an issue to do it later? 🙇

Reply:
Hmm, not sure what a style guide would look like. Do you have any proposals or good and short examples? 

And https://pub.dev/packages/very_good_analysis covers styling lint rules.

Reply:
If we're using the default dart style guide that's being enforced by the `very_good_analysis` then let's just specify it here 😉 To me "No particular style guide" sounds like we don't care about code style and that's not true 😅

Reply:
Updated in https://github.com/nomtek/mistralai_client_dart/pull/24/commits/42354570f194dd5e88f3982f0b824e9e1bbca23b